src/lightning/fabric/utilities/warnings.py

```diff
@@ -12,13 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Warning-related utilities."""
+import importlib.util
+import os
 import warnings
+from functools import wraps
+from pathlib import Path
+from typing import Callable, List, Optional, Type, Union
 
 from lightning.fabric.utilities.rank_zero import LightningDeprecationWarning
 
 # enable our warnings
 warnings.simplefilter("default", category=LightningDeprecationWarning)
 
 
+def _wrap_formatwarning(default_format_warning: Callable) -> Callable:
+    """Custom formatting that avoids an extra line in case warnings are emitted from the `rank_zero`-functions."""
+
+    @wraps(default_format_warning)
+    def wrapper(
+        message: Union[Warning, str], category: Type[Warning], filename: str, lineno: int, line: Optional[str] = None
+    ) -> str:
+        if _is_path_in_lightning(Path(filename)):
+            # The warning originates from the Lightning package
+            return f"{filename}:{lineno}: {message}\n"
+        return default_format_warning(message, category, filename, lineno, line)
+
+    return wrapper
+
+
+warnings.formatwarning = _wrap_formatwarning(warnings.formatwarning)
+
+
 class PossibleUserWarning(UserWarning):
     """Warnings that could be false positives."""
+
+
+def _is_path_in_lightning(path: Path) -> bool:
+    """Checks whether the given path is a subpath of the Lightning package."""
+    path = Path(path).absolute()
+    lightning_roots = _get_lightning_package_roots()
+    for lightning_root in lightning_roots:
+        if path.drive != lightning_root.drive:  # handle windows
+            continue
+        common_path = Path(os.path.commonpath([path, lightning_root]))
+        if common_path.name in ("lightning", "lightning_fabric", "pytorch_lightning"):
+            return True
+    return False
```

Comment:
Why is this function this complicated? Why not simply do `return "lightning" in str(path)`?

Reply:
What is complicated about the function? Is there a way to make it simpler?
I did this to implement your earlier comment about only applyting the formatting for when the warning originates from our code. This function does it. 
The `"lightning" in str(path)` would obviously not be reliable as it only checks whether the string lightning appears anywhere in the path.

Reply:
I'm just saying that this is unnecessarily complicated. A substring check should be sufficient in most cases. Looking at the test you added it would cover everything but the case where the filesystem is different, but I wouldn't worry about that.

This will run on every warning, so if you want to keep this version, cache `_get_lightning_package_roots` at least